profiling/src/php_ffi.c

```diff
@@ -207,6 +207,15 @@ static bool has_invalid_run_time_cache(zend_function const *func) {
     if (UNEXPECTED(_ignore_run_time_cache) || UNEXPECTED(ddog_php_prof_run_time_cache_handle < 0))
         return true;
 
+    // during an `include()`/`require()` with enabled OPcache, OPcache is
+    // persisting the compiled file and puts a fake frame on the stack where the
+    // runtime cache is not yet initialized.
+#if PHP_VERSION_ID < 80200
+    bool is_file_compile = func->op_array.run_time_cache__ptr == NULL;
```

Comment:
I don't think we're supposed to reach into that. @bwoebi , is there a macro or function you are aware of?

Side note: if the engine is making fake frames, it should at least make complete frames...

Reply:
```suggestion
    bool is_file_compile = ZEND_MAP_PTR(func->op_array.run_time_cache) == NULL;
```
would be the proper usage